lib/src/textboxGroup/textboxGroup.component.ts

```diff
@@ -0,0 +1,153 @@
+import {
+    Component,
+    Input,
+    ViewEncapsulation,
+    ElementRef,
+    forwardRef,
+    Provider,
+    OnChanges,
+    SimpleChanges,
+    OnInit,
+    Output,
+    EventEmitter,
+} from "@angular/core";
+import { randomId } from "@sebgroup/frontend-tools/dist/randomId";
+
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+
+const CUSTOM_DROPDOWN_CONTROL_VALUE_ACCESSOR: Provider = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => TextboxGroupComponent),
+    multi: true,
+};
+
+type Value = number | string;
+
+@Component({
+    selector: "sebng-textboxgroup",
+    templateUrl: "./textboxGroup.component.html",
+    styleUrls: ["./textboxGroup.component.scss"],
+    encapsulation: ViewEncapsulation.None,
+    providers: [CUSTOM_DROPDOWN_CONTROL_VALUE_ACCESSOR],
+})
+export class TextboxGroupComponent implements ControlValueAccessor, OnInit, OnChanges {
+    @Input() autoComplete?: "on" | "off";
+    @Input() className?: string;
+    @Input() disabled?: boolean;
+    @Input() error?: string;
+    @Input() focus?: boolean;
+    @Input() id?: string;
+    @Input() label?: string;
+    @Input() leftIcon?: string;
+    @Input() leftText?: string;
+    @Input() leftTitle?: string;
+    @Input() maxLength?: number;
+    @Input() minLength?: number;
+    @Input() name: string;
+    @Input() pattern?: string;
+    @Input() placeholder?: string;
+    @Input() readOnly?: boolean;
+    @Input() required?: boolean;
+    @Input() rightIcon?: string;
+    @Input() rightText?: string;
+    @Input() rightTitle?: string;
+    @Input() type?: string;
+    @Input() success?: boolean;
+    @Input() showErrorMessage?: boolean;
+
+    @Output() onBlur = new EventEmitter<MouseEvent>();
+    @Output() onChange = new EventEmitter<Value>();
+    @Output() onFocus = new EventEmitter<MouseEvent>();
+    @Output() onKeyDown = new EventEmitter<KeyboardEvent>();
+    @Output() onKeyPress = new EventEmitter<KeyboardEvent>();
+    @Output() onKeyUp = new EventEmitter<KeyboardEvent>();
+    @Output() onLeftClick = new EventEmitter<MouseEvent>();
+    @Output() onRightClick = new EventEmitter<MouseEvent>();
+
+    private innerValue: Value;
+
+    private onTouchedCallback: () => void;
+    private onChangeCallback: (_: any) => void;
+
+    public internalId: string;
+    public displayError: boolean;
+
+    ngOnInit() {
+        this.internalId = this.id ? this.id : this.label ? randomId("tbg-") : null;
```

Comment:
you are forcing the user to have an id, i dont think you should do that and instead use @Input() id directly in the component template

Reply:
I wouldn't say forcing the user, we are putting one by default for him incase he chose not to. I couldn't remember what the discussion was , but we had a discussion last time and decided to go with ramdom id for specific input elements of which texboxGroup was among.  I had remove it here, but i couldn't remember  what we address with it in react-component. We might have to discuss again perhaps.

Reply:
I believe the random ID serves one purpose here, enabling a feature where clicking on the label will focus on the input. I don't think there is an issue with that.